pkg/gui/branches_panel.go

```diff
@@ -9,154 +9,346 @@ import (
 	"github.com/jesseduffield/lazygit/pkg/git"
 )
 
+// gui.refreshStatus is called at the end of this because that's when we can
+// be sure there is a state.Branches array to pick the current branch from
+func (gui *Gui) refreshBranches() error {
+
+	gui.g.Update(func(g *gocui.Gui) error {
+
+		v, err := gui.g.View("branches")
+		if err != nil {
+			gui.Log.Errorf("Failed to get branches view at refreshbranches %s\n", err)
+			return err
+		}
+
+		builder, err := git.NewBranchListBuilder(gui.Log, gui.GitCommand)
+		if err != nil {
+			gui.Log.Errorf("Failed to create branchbuilder at refreshBranches: %s\n", err)
+			return err
+		}
+
+		gui.State.Branches = builder.Build()
+
+		v.Clear()
+
+		for _, branch := range gui.State.Branches {
+			fmt.Fprintln(v, branch.GetDisplayString())
+		}
+
+		err = gui.resetOrigin(v)
+		if err != nil {
+			gui.Log.Errorf("Failed to reset origin at refreshBranches: %s\n", err)
+			return err
+		}
+
+		err = gui.refreshStatus()
+		if err != nil {
+			gui.Log.Errorf("Failed to refresh statsu at refreshBranches: %s\n", err)
+			return err
+		}
+
+		return nil
+	})
+
+	return nil
+}
+
+// handleBranchPress is called when the user selects a branch.
+// g and v are passed by the gocui library, but are not used.
+// In case something goes wrong it returns an error
 func (gui *Gui) handleBranchPress(g *gocui.Gui, v *gocui.View) error {
-	index := gui.getItemPosition(gui.getBranchesView(g))
+
+	v, err := gui.g.View("branches")
+	if err != nil {
+		gui.Log.Errorf("Failed to get branch view at handleBranchPress: %s\n", err.Error())
+	}
+
+	index := gui.getItemPosition(v)
 	if index == 0 {
-		return gui.createErrorPanel(g, gui.Tr.SLocalize("AlreadyCheckedOutBranch"))
+
+		err := gui.createErrorPanel(gui.Tr.SLocalize("AlreadyCheckedOutBranch"))
+		if err != nil {
+			gui.Log.Errorf("Failed to create error panel at handleBranchPress: %s\n", err)
+			return err
+		}
+
+		return nil
 	}
-	branch := gui.getSelectedBranch(gui.getBranchesView(g))
-	if err := gui.GitCommand.Checkout(branch.Name, false); err != nil {
-		gui.createErrorPanel(g, err.Error())
+
+	branch := gui.getSelectedBranch(v)
+
+	err = gui.GitCommand.Checkout(branch.Name, false)
```

Comment:
I think we should keep `if err := gui.GitCommand.Checkout(branch.Name, false); err != nil {` here as it's more expressive and reads more easily

Reply:
For the sake of consistency I applied one style everywhere. This because I didn't want to introduce multiple styles.

Reply:
Would it be considered multiple styles to always use the `; err != nil {` syntax when the function only returns an error, but use separate lines when other values are returned? I would not consider that as two separate styles because in the former situation there is the opportunity to be more more expressive by restricting the scope of the err variable.

Nonetheless I won't let that hold up this PR

Reply:
True, ah well ill let this stay for now, since well i'm exhausted after 3.5k lines diff lol. Maybe we'll get to it when/if there comes a part 2